### PR TITLE
Use a Makefile

### PR DESCRIPTION
**Pull request by zprobst**
This is intended to provide some handy short-cuts and utility commands for working with the repo.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+format: venv
+	poetry run black nodestream tests
+	poetry run isort nodestream tests
+
+.PHONY: lint
+lint: venv
+	poetry run black nodestream tests --check
+	poetry run ruff nodestream tests
+
+.PHONY: test
```
Comment (ccloes) [inline Makefile:38]: Should this be test-unit?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+test-unit: venv
+	poetry run pytest -m "not e2e"
+
+.PHONY: e2e
```
Comment (ccloes) [inline Makefile:42]: Should this be test-e2e?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+test-e2e: venv
+	poetry run pytest -m "e2e"
+
+.PHONY: docs
+docs: docs-build docs-publish
```
Comment (ccloes) [inline Makefile:47]: Missing docs-publish ...

**Review by ccloes — APPROVED** [review]
A couple of tweaks I think you need... but approved anyway.  `make snapshot` didn't work for me either.

**Review by grantleehoffman — APPROVED** [review]

**Pull request merged**